### PR TITLE
feat: Allowing password based authentication and SSL for Redis in Java feature server

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
This is a partial forward port of https://github.com/feast-dev/feast-java-old/pull/44 to enable auth for Redis in the Java feature server (Python already handles this)

**Which issue(s) this PR fixes**:

Fixes # https://github.com/feast-dev/feast/issues/2285

**Does this PR introduce a user-facing change?**:

```release-note
Java feature server now supports SSL and password based authentication for Redis
```

### CODE REVIEW

File: java/serving/src/main/java/feast/serving/config/ApplicationProperties.java [hunk 1]
```diff
@@ -246,10 +249,30 @@ public void setType(String type) {
...
+
+      if (!this.config.containsKey("timeout") || this.config.get("timeout") == null) {
+        throw new RuntimeException("Redis cluster config does not have 'timeout' specified");
+      }
+
+      boolean ssl = false;
+      if (!this.config.containsKey("ssl") || this.config.get("ssl") == null) {
+        log.info("'ssl' not defined in Redis cluster config, so setting to false");
+      } else {
+        ssl = Boolean.parseBoolean(this.config.getOrDefault("ssl", "false"));
```
Comment (woop) [inline java/serving/src/main/java/feast/serving/config/ApplicationProperties.java:267]: In which case would the default be used?

**Review by andrijaperovic — COMMENTED** [review]
@adchia do you plan to port the tcp keepalive and socket resolver changes as well? Or you want me to add these changes separately.

**Comment by adchia** [comment]
> @adchia do you plan to port the tcp keepalive and socket resolver changes as well? Or you want me to add these changes separately.

@andrijaperovic yeah can you make the other changes too? Wanted to get this one in first since it's blocking some users

**Review by andrijaperovic — CHANGES_REQUESTED** [review]

**Comment by adchia** [comment]
friendly bump @woop

File: java/serving/src/main/java/feast/serving/config/ApplicationProperties.java [hunk 2]
```diff
@@ -246,10 +249,31 @@ public void setType(String type) {
      * @return Returns the store specific configuration
      */
     public RedisClusterStoreConfig getRedisClusterConfig() {
+      String read_from = ReadFrom.UPSTREAM.toString();
+      if (!this.config.containsKey("read_from") || this.config.get("read_from") == null) {
+        log.info("'read_from' not defined in Redis cluster config, so setting to UPSTREAM");
```
Comment (woop) [inline java/serving/src/main/java/feast/serving/config/ApplicationProperties.java:254]: Sorry this is a bit nitpicky, but isnt it safer to have the variable be set within the `if` statement? It's easier to read, and less likely to be refactored erroneously. Same for the below if statement.

Comment (adchia) [reply java/serving/src/main/java/feast/serving/config/ApplicationProperties.java:254]: Done

**Pull request merged**